pkg/engine/common/pattern.go

```diff
@@ -285,11 +285,12 @@ func validateNumberWithStr(log logr.Logger, value interface{}, pattern string, o
 	}
 
 	// 2. wildcard match
-	if !wildcard.Match(pattern, typedValue) {
+	if validateString(log, value, pattern, operator) {
```

Comment:
@4molybdenum2 Thanks for this PR! Could you also please add a test case under CLI tests with your proof manifests to ensure that this is tested automatically and we can catch any regressions in the future?

Reply:
Added tests

Reply:
That and you can also check in the proof manifests you have at https://github.com/kyverno/kyverno/tree/main/test/cli/test under any of the sub-folders or create a new one as appropriate. For more details see https://kyverno.io/docs/kyverno-cli/#test for the file formats. We automatically call kyverno test on that sub-folder so you just need to add an appropriate `kyverno-test.yaml` file or add an appropriate policy and resource and reference it in an existing test file.

Reply:
@4molybdenum2 could you add CLI tests as well please ^ thanks!

Reply:
@samj1912 I'm not exactly sure how we can use CLI tests to verify this policy (as it gets the status from the Node to validate directly...) but I can add the check kernel version policy under the tests folder?

Reply:
Either that or you can get the same result if you try and validate a field that accepts type string and using a numerical value like the type outlined in the issue.

Reply:
Added tests for the following:
1. Validate string with numerical value and negation
2. Validate multiple patterns with negations joined using & operator

Also added the check-kernel policy under test/policy/validate

Reply:
Can you add CLI test cases that correctly pass/fail given an appropriate resource?

The current policy that you have checked in just means that kyverno will check if that is a valid policy or not. I believe the @chipzoller noted in the issue that the policy was failing in cases it should have passed.

Reply:
@samj1912 added the CLI test cases too....
Its working as expected